src/main/java/org/apache/ibatis/reflection/Reflector.java

```diff
@@ -140,35 +144,26 @@ private void addSetMethods(Class<?> clazz) {
   }
 
   private void addMethodConflict(Map<String, List<Method>> conflictingMethods, String name, Method method) {
-    List<Method> list = conflictingMethods.computeIfAbsent(name, k -> new ArrayList<>());
-    list.add(method);
+    if (isValidPropertyName(name)) {
+      List<Method> list = conflictingMethods.computeIfAbsent(name, k -> new ArrayList<>());
+      list.add(method);
+    }
   }
 
   private void resolveSetterConflicts(Map<String, List<Method>> conflictingSetters) {
     for (String propName : conflictingSetters.keySet()) {
       List<Method> setters = conflictingSetters.get(propName);
       Class<?> getterType = getTypes.get(propName);
       Method match = null;
-      ReflectionException exception = null;
       for (Method setter : setters) {
         if (setter.getParameterTypes()[0].equals(getterType)) {
           // should be the best match
           match = setter;
           break;
         }
-        if (exception == null) {
-          try {
-            match = pickBetterSetter(match, setter, propName);
-          } catch (ReflectionException e) {
-            // there could still be the 'best match'
-            match = null;
-            exception = e;
-          }
-        }
+        match = pickBetterSetter(match, setter, propName);
```

Comment:
Probably we should control to not call the `pickBetterSetter` method after detect ambiguous setter methods(=I think should find only best match method after detect ambiguous setter methods).
In changed implementation, for example If there are three conflict setter methods, the last method become as better method and will invoke it. If my understanding is correct, I think that should throw the `ReflectionException` instead of calling setter method in this case . WDYT?

Reply:
Good catch!  I have added two commits.
Please let me know if you noticed something (no rush, of course).